### ISSUE TITLE
Configure peerDependencies to avoid duplicated React instances in bundles

### ISSUE THREAD

**Issue by sarimarton** [cyclejs/react#2]
React instances are duplicated when using cycle/react, because cycle/react defines react as a regular dependency instead of a peerDependency in package.json. This not only bloats the bundle files up, but also makes writing extensions harder, because in some low-level cases, these instances can mix up (especially when I'm playing with JSX pragmas). Currently these are regular dependencies package.json:
```
  "dependencies": {
    "@cycle/run": "5.x.x",
    "react": "16.5.x",
    "xstream": "11.x.x"
  },
```
I think all of these should be in the peerDependencies block.

**Comment by sarimarton** [comment]
It's probably not directly a package.json issue. The issue is that cyclejs/react will always pick up its own react instance located in its node_modules folder published with the package - which in turn causes a double instance of React in the bundle file, and it leads to different spooky interference issues. So a PR wouldn't help much here.

**Comment by staltz** [comment]
I think if we choose a wide enough range for `react` as a dependency, then npm would not have duplicates. At least that's been my experience when working with package-lock.json, I hope this hasn't changed.

**Comment by sarimarton** [comment]
I'm not sure that that would approach the problem from the right angle. That would be just a partial solution for a theorietically wrong config. This is a book example for what peer dependencies were made: to guarantee that a certain dependency is not doubled. Why not going with peer dependencies?

This is currently a blocker for me, because I can't use React components which use hooks with cyclejs/react. I'm about to introduce cyclejs in my strict enterprise company, and I need to demonstrate that cyclejs/react plays well in a modern react codebase (which has hooks).

**Comment by staltz** [comment]
Fixed, published 2.2.0
